### ISSUE TITLE
Weird error about variable not being defined when using &&

### ISSUE THREAD

**Issue by nh2** [xonsh/xonsh#2430]
Does anybody know what's going on here?

Note, if you can't reproduce, I suspect it's something in my tree, but I haven't found anything yet so I'm first asking if it happens for you, too.

```
$ rm -f a && make && ./c
xonsh: For full traceback set: $XONSH_SHOW_TRACEBACK = True
NameError: name 'make' is not defined
```

```
$ rm -f a && make && ./c
xonsh: To log full traceback to a file set: $XONSH_TRACEBACK_LOGFILE = <filename>
Traceback (most recent call last):
  File "/home/niklas/python35env/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 15472, in default
    run_compiled_code(code, self.ctx, None, 'single')
  File "/home/niklas/python35env/lib/python3.5/site-packages/xonsh/__amalgam__.py", line 3634, in run_compiled_code
    func(code, glb, loc)
  File "/home/niklas/src/xontrib-fzf-widgets/xontrib/fzf-widgets.xsh", line 1, in <module>
    import os
NameError: name 'make' is not defined
```

```
$ xonfig
+------------------+----------------------+
| xonsh            | 0.5.12.dev4          |
| Git SHA          | 06c57eec             |
| Commit Date      | Jun 30 13:56:45 2017 |
| Python           | 3.5.2                |
| PLY              | 3.9                  |
| have readline    | True                 |
| prompt toolkit   | 1.0.14               |
| shell type       | prompt_toolkit       |
| pygments         | 2.2.0                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | Ubuntu               |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
+------------------+----------------------+
```

**Comment by astronouth7303** [comment]
Yup, I can reproduce on v0.5.12 with:

```
true && false && true
```

@scopatz Sounds like a parser bug?

**Comment by scopatz** [comment]
Thanks for reporting.  @astronouth7303  it isn't a parser bug because the code is being compiled and executed. It is simply failing to autowrap the second command for some reason

**Comment by gforsyth** [comment]
Closed in #2845
